sherpa/utils/testing.py

```diff
@@ -149,10 +149,13 @@ def requires_pylab(test_function):
         otherwise skips the test.
         """
         pylab_backend = pytest.importorskip("sherpa.plot.pylab_backend")
+        plt  = pytest.importorskip("matplotlib.pyplot")
 
         with TemporaryPlottingBackend(pylab_backend.PylabBackend()):
             yield
 
+        plt.close(fig="all")
```

Comment:
Unfortunately not. I run it within an iython session:
```Python
In [1]: run -m pytest
In [2]: import matplotlib
In [3]: matplotlib.pyplot.get_fignums()
Out[3]: []
```

If I remove the `cleanup_pylab_backend` fixture, all tests still pass, but there is an open window left. I'm not quite sure, but I suspect that that's from a doctest. It could be dealt with in some other way (e.g. tear-down code after the doctest), but this fixture works and does not require us to pollute other files with tear-down code, so I suggest to just leave it in.

Reply:
Okay. Worth checking.